driver-core/src/main/java/com/datastax/driver/core/AbstractSession.java

```diff
@@ -51,6 +51,7 @@ public ResultSet execute(String query, Object... values) {
      */
     @Override
     public ResultSet execute(Statement statement) {
+        checkNotInEventLoop();
```

Comment:
Not sure we should do the check in `AbstractSession`. I understand the motivation to make all possible subclasses resilient to IO thread misuse but 1) users will have to implement `checkNotInEventLoop` anyway (and I bet they won't) 2) some implementations may not even use Netty at all, in which case the method would sound inappropriate. Why not simply keep this check internal to `SessionManager`?